File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/model/converter/SentInviteConverter.java
package com.worldventures.dreamtrips.social.ui.membership.model.converter;

import com.worldventures.core.converter.Converter;
import com.worldventures.dreamtrips.api.invitation.model.Invitation;
import com.worldventures.dreamtrips.social.ui.membership.model.SentInvite;

import io.techery.mappery.MapperyContext;

public class SentInviteConverter implements Converter<Invitation, SentInvite> {

   @Override
   public Class<Invitation> sourceClass() {
      return Invitation.class;
   }

   @Override
   public Class<SentInvite> targetClass() {
      return SentInvite.class;
   }

   @Override
   public SentInvite convert(MapperyContext mapperyContext, Invitation invitation) {
      return new SentInvite(invitation.contact(), invitation.date());
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/domain/converter/SentInviteConverter.kt
package com.worldventures.dreamtrips.social.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.invitation.model.Invitation
import com.worldventures.dreamtrips.social.domain.entity.SentInvite

import io.techery.mappery.MapperyContext

class SentInviteConverter : Converter<Invitation, SentInvite> {

   override fun sourceClass() = Invitation::class.java

   override fun targetClass() = SentInvite::class.java

   override fun convert(context: MapperyContext, source: Invitation) = SentInvite(source.contact(), source.date())
}
